### ISSUE TITLE
Boxplot as a feature?

### ISSUE THREAD

**Issue by DooRooSA** [imaNNeoFighT/fl_chart#934]
Good afternoon,

I use Excel's Boxplot charts quite often in managing my business/life. I would love to see Boxplot charts in fl_chart, as I have numerous places I can use it in my own projects.

Although I've been a solo developer for as long as I can remember, I've never contributed to an open-source project.

I would like to tackle this feature, but would not want to tread on any toes. Is there somebody who can guide me through the process of adding Boxplot charts to this project?

**Comment by imaNNeoFighT** [comment]
Hi @DooRooSA 
Contribution can be started simply. 
For example, now you are contributing to this project by mentioning your issue (demanding a new chart). 
Yes trust me, this is a contribution!

BTW, start by reading our [CONTRIBUTING](https://github.com/imaNNeoFighT/fl_chart/blob/master/CONTRIBUTING.md) guideline
Your next step can be finding a simple issue to fix, I tried to group some simple issues [here](https://github.com/imaNNeoFighT/fl_chart/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22).

Then start reading our [ScatterChart](https://github.com/imaNNeoFighT/fl_chart/blob/master/repo_files/documentations/scatter_chart.md) source code. It's our simplest cartesian-based chart.

**Comment by imaNNeoFighT** [comment]
Anyways it seems like a candle stick chart. Correct?
If yes, there is another open issue here: #433. 
We can close it and follow it up there.

**Comment by DooRooSA** [comment]
Thank  you, I will certainly look at starting on some of the simpler issues first to get used to the process.

As for the boxplot vs candlestick, they are certainly different charts altogether. As I understand the candlestick is used for trading financial instruments. The boxplot is a statistical analysis chart that shows the median, quartiles and maximum and minimum values of a dataset. It can also be configured to display the outliers, which is very useful to see. The boxplot shines in the fact that you can quite easily see if your data is skewed.

Although they look similar, their purpose is vastly different. I suppose the boxplot can also be grouped in a similar way that the BarChart is grouped on fl_chart (possible code re-use?). The image below is the format that I use 99% of the time, so this format would be a good starting point in my opinion, as it encompasses most of the features of a boxplot.

![32937494-baab4626-cbc8-11e7-8660-f4a85ed7254b](https://user-images.githubusercontent.com/17521217/156971299-95fa5cab-1b2b-4e97-8ec1-9053767ac7aa.jpg)

**Comment by imaNNeoFighT** [comment]
I'm thinking of having a generic CandleStickc chart that supports both usecases.
They have a lot of shared things to draw.
What are those dots?

**Comment by DooRooSA** [comment]
I agree, the visuals are very similar. 

The dots are the outliers. I'm including a screenshot of Excel that I made to explain them. Here is a list of numbers between 20 and 30. The last 3 numbers, fall outside those two values. I've marked them in orange in the table. You will notice that the values 8, 6 and 41 are indicated as outliers on the chart

![Screenshot 2022-03-10 202415](https://user-images.githubusercontent.com/17521217/157730825-57c983e1-8729-4d9e-8d00-38d42bc6eb12.png)
